src/site/content/en/blog/hands-on-portals/index.md

```diff
@@ -0,0 +1,203 @@
+portal.postMessage({someKey: someValue}, ORIGIN);
+
+// Receive message via window.portalHost
+window.portalHost.addEventListener('message', evt => {
+  const data = evt.data.someKey;
+  // handle the event
+});
+```
+### Activating the portal element and receiving the `portalactivate` event
+```javascript
+// You can optionally add data to the argument of the activate function
+portal.activate({data: {'somekey': 'somevalue'}});
+
+// The portal content will receive the portalactivate event
+// when the activate happens
+window.addEventListener('portalactivate', evt => {
+  // Data available as evt.data
+  const data = evt.data;   
+});
+```
+### Retrieving the predecessor
+```javascript
+// Listen to the portalactivate event
+window.addEventListener('portalactivate', evt => {
+  // ... and creatively use the predecessor
+  const portal = evt.adoptPredecessor();
+  document.querySelector('someElm').appendChild(portal);
+});
+```
+### Knowing your page was adopted as a predecessor
+```javascript
+// The activate function returns a Promise.
+// When the promise resolves, it means that the portal has been activated.
+// If this document was adopted by it, then window.portalHost will exist.
+portal.activate().then(_ => {
+  // Check if this document was adopted into a portal element.
+  if (window.portalHost) {
+    // You can start communicating with the portal element
+    // i.e. listen to messages
+    window.portalHost.addEventListener('message', evt => {
+      // handle the event
+    });
+  }
+});
+```
+By combining all the features supported by Portals, you can build really fancy user experiences. For instance the demo below demonstrates how Portals can enable a seamless user experience between a website and a third party embed content. 
+
+<div style="width:100%; padding-top: 56.25%; position: relative;">
+  <iframe style="width:100%; height: 100%;position: absolute; top: 50%; left: 50%; transform: translate(-50%,-50%);" src="https://www.youtube.com/embed/4JkipxFVE9k" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+</div>
+
+[Here is the link to the GitHub repo](https://github.com/WICG/portals/tree/master/demos/portal-embed-demo) which you can play around with. Be creative and have fun!!
+
+## Use cases and the plan
+I hope you liked this brief tour of Portals! We can’t wait to see what you can come up with. For instance, you might want to start using Portals for non-trivial navigations such as: pre-rendering the page for your best-seller product from a product category listing page. Another important thing to know is that Portals can be used in cross-origin navigations, just like iframes. So, if you have multiple websites that cross reference one another, you can also use Portals to create seamless navigations between two different websites. This cross-origin use case is very unique to Portals, and can even improve the user experience of SPAs.
```

Comment:
Start new paragraph with "Another important thing to know".

Reply:
Thank you! Will fix!!